### PR TITLE
make bbc-a11y run PGL pages

### LINKED ISSUE #5411: Run bbc-a11y on PGLs during build
**Is your feature request related to a problem? Please describe.**
Currently we aren’t running `bbc-a11y` on CI/CD for any PGL assets. This increases the chance we could accidentally make photo galleries not accessible.

**Describe the solution you'd like**
- Ensure that this tool is run on one PGL and that any failures cause the build to fail.
- The bbc-a11y module uses `./bbcA11y.js` to configure its run. Add the PGL page type here.

Note:
- Investigate whether this test should also be run against `test` - if so, another issue should be raised to amend this file `simorgh-infrastructure/config/articles.txt`.

**Describe alternatives you've considered**
None.

**Testing notes**
- Run `npm run dev && npm run bbcA11y` to determine a11y tests are being run on PGL assets.
- Also verify it is being run correctly on CI and will cause the build to fail.

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Additional context**
- https://github.com/bbc/bbc-a11y

### PR DESCRIPTION
**Pull request by rhenshaw56**
Resolves #5411 

**Overall change:** 
Allows bbc-a11y run PGL page types.

**Code changes:**
- Added page type of`photoGalleryPage` (referenced from cypress service config) to a11y.js
---

- [X] I have assigned myself to this PR and the corresponding issues
- [X] I have added labels to this PR for the relevant pod(s) affected by these changes
- [X] I have assigned this PR to the Simorgh project

**Testing:**
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by 12 — COMMENTED** [review]
Do we not need to add a PGL asset to the commands in the `package.json`?

https://github.com/bbc/simorgh/blob/4aad8885d5d1304b169fcabc15a8e89a0c496fe5/package.json#L8-L9

File: a11y.js [hunk 1]
```diff
@@ -37,6 +37,7 @@ const pageTypes = {
     '/html/head/iframe',
     "//div[@id='root']/main/div/div/div/div/iframe",
   ],
+  photoGalleryPage: ['/html/head/iframe'],
```
Comment (andrewscfc) [inline a11y.js:40]: I don't think this is wrong, but what does `'/html/head/iframe'` do? It seems to mention some exclusion for mpulse above?

**Comment by rhenshaw56** [comment]
> Do we not need to add a PGL asset to the commands in the `package.json`?
> 
> https://github.com/bbc/simorgh/blob/4aad8885d5d1304b169fcabc15a8e89a0c496fe5/package.json#L8-L9

@12 I thought so initially but doing that doesn't run bbc-a11y tests for PGLs. 
It seems `wait-on -t 20000 http://localhost:7080/news/articles/cn7k01xp8kxo` just ensures that the specified url is available before running the `bbc-a11y` command, which looks out for an `a11y.js` file and runs the urls specified there. 
From their [docs](https://github.com/bbc/bbc-a11y/blob/master/guides/using/using-bbc-a11y-in-your-project.md#add-a11y-to-your-projects-dependencies)
> You'll need to configure a11y with a set of URLs to run the checks against. Create a file called a11y.js in the root of your project that looks something like this:
```
page("http://bbc.co.uk")
page("http://bbc.co.uk/news")
```

File: a11y.js (hunk 1 above)
Comment (rhenshaw56) [reply a11y.js:40]: yeah @andrewscfc I think so too and also from the [bbc-a11y docs](https://github.com/bbc/bbc-a11y/blob/master/guides/using/using-bbc-a11y-in-your-project.md#hiding-errors) there's a section about doing this to hide specific errors from elements which are bound to fail a11y checks

**Comment by andrewscfc** [comment]
> > Do we not need to add a PGL asset to the commands in the `package.json`?
> > https://github.com/bbc/simorgh/blob/4aad8885d5d1304b169fcabc15a8e89a0c496fe5/package.json#L8-L9
> 
> @12 I thought so initially but doing that doesn't run bbc-a11y tests for PGLs.
> It seems `wait-on -t 20000 http://localhost:7080/news/articles/cn7k01xp8kxo` just ensures that the specified url is available before running the `bbc-a11y` command, which looks out for an `a11y.js` file and runs the urls specified there.
> From their [docs](https://github.com/bbc/bbc-a11y/blob/master/guides/using/using-bbc-a11y-in-your-project.md#add-a11y-to-your-projects-dependencies)
> 
> > You'll need to configure a11y with a set of URLs to run the checks against. Create a file called a11y.js in the root of your project that looks something like this:
> 
> ```
> page("http://bbc.co.uk")
> page("http://bbc.co.uk/news")
> ```

Oh so it's waiting for arbirary url be available before running all the a11y tests? I guess once that url is available, they all will be?

File: a11y.js (hunk 1 above)
Comment (andrewscfc) [reply a11y.js:40]: Ah makes sense looking again with the docs, looks good

**Comment by rhenshaw56** [comment]
> > > Do we not need to add a PGL asset to the commands in the `package.json`?
> > > https://github.com/bbc/simorgh/blob/4aad8885d5d1304b169fcabc15a8e89a0c496fe5/package.json#L8-L9
> > 
> > 
> > @12 I thought so initially but doing that doesn't run bbc-a11y tests for PGLs.
> > It seems `wait-on -t 20000 http://localhost:7080/news/articles/cn7k01xp8kxo` just ensures that the specified url is available before running the `bbc-a11y` command, which looks out for an `a11y.js` file and runs the urls specified there.
> > From their [docs](https://github.com/bbc/bbc-a11y/blob/master/guides/using/using-bbc-a11y-in-your-project.md#add-a11y-to-your-projects-dependencies)
> > > You'll need to configure a11y with a set of URLs to run the checks against. Create a file called a11y.js in the root of your project that looks something like this:
> > 
> > 
> > ```
> > page("http://bbc.co.uk")
> > page("http://bbc.co.uk/news")
> > ```
> 
> Oh so it's waiting for arbirary url be available before running all the a11y tests? I guess once that url is available, they all will be?

@andrewcfc yes

**Review by simonsinclair — COMMENTED** [review]
Nice work, Rowland. Have we confirmed whether the build fails if bbc-a11y exits with an error?

**Comment by rhenshaw56** [comment]
> Nice work, Rowland. Have we confirmed whether the build fails if bbc-a11y exits with an error?

@simonsinclair yeah, It should fail.

**Comment by simonsinclair** [comment]
> > Nice work, Rowland. Have we confirmed whether the build fails if bbc-a11y exits with an error?
> 
> @simonsinclair yeah, It should fail.

I think it's important to confirm that it _does_ fail before this is merged. Have we confirmed that? If not, we can do this by breaking accessibility on a PGL asset and testing locally with `npm run bbcA11y`, then pushing the change to see if it breaks when `runProductionTests()` are executed on this branch in CI.

**Comment by rhenshaw56** [comment]
> > > Nice work, Rowland. Have we confirmed whether the build fails if bbc-a11y exits with an error?
> > 
> > 
> > @simonsinclair yeah, It should fail.
> 
> I think it's important to confirm that it _does_ fail before this is merged. We can do this by breaking accessibility on a PGL asset and testing locally with `npm run bbcA11y`, then pushing the change to see if it breaks when `runProductionTests()` are executed on this branch in CI.

@simonsinclair yeah, removing the alt attribute from the image element results in failures
Screen Shot 2020-02-13 at 14 40 53

**Review by simonsinclair — APPROVED** [review]

**Review by andrewscfc — APPROVED** [review]
Thanks for explaining! 🚢

**Pull request merged**